felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -529,6 +528,32 @@ func newBPFEndpointManager(
 		features:         dataplanefeatures,
 	}
 
+	specialInterfaces := []string{"egress.calico"}
+	if config.RulesConfig.IPIPEnabled {
+		specialInterfaces = append(specialInterfaces, dataplanedefs.IPIPIfaceName)
+	}
+	if config.RulesConfig.VXLANEnabled {
+		specialInterfaces = append(specialInterfaces, dataplanedefs.VXLANIfaceNameV4)
+	}
+	if config.RulesConfig.VXLANEnabledV6 {
+		specialInterfaces = append(specialInterfaces, dataplanedefs.VXLANIfaceNameV6)
+	}
+	if config.RulesConfig.WireguardEnabled {
+		specialInterfaces = append(specialInterfaces, config.RulesConfig.WireguardInterfaceName)
+	}
+	if config.RulesConfig.WireguardEnabledV6 {
+		specialInterfaces = append(specialInterfaces, config.RulesConfig.WireguardInterfaceNameV6)
+	}
+
+	exp := "(" + config.BPFDataIfacePattern.String() + ")|("
+	for _, d := range specialInterfaces {
+		exp += "^" + d + "$|"
+	}
```

Comment:
```suggestion
	for i, d := range specialInterfaces {
		specialInterfaces[i] = "^" + regexp.Quote(d) + "$"
	}
	exp := config.BPFDataIfacePattern.String() + "|" + strings.Join(specialInterfaces, "|")
```
Recommend quoting the names (since some have dots in? `strings.Join` is a nice way to deal with the delimiter.